scm/etc/modules/derecho_intel.lua

```diff
@@ -0,0 +1,47 @@
+help([[
+This module loads libraries for building the CCPP Single-Column Model on
+the CISL machine Derecho (Cray) using Intel-classic-2023.0.0
+]])
+
+whatis([===[Loads libraries needed for building the CCPP SCM on Derecho ]===])
+
+load(pathJoin("cmake", os.getenv("cmake_ver") or "3.26.3"))
+load(pathJoin("ncarenv", os.getenv("ncarenv_ver") or "23.06"))
+load(pathJoin("craype", os.getenv("craype_ver") or "2.7.20"))
+
+unload("netcdf")
+unload("hdf5")
+load(pathJoin("intel-classic", os.getenv("intel_classic_ver") or "2023.0.0"))
+load(pathJoin("cray-mpich", os.getenv("cray_mpich_ver") or "8.1.25"))
+
+prepend_path("MODULEPATH","/glade/work/epicufsrt/contrib/derecho/hpc-stack/intel-classic-2023.0.0/modulefiles/stack")
```

Comment:
I think that there is still a reliance on some non-standard python packages, like f90nml. Would we still need to create a python environment on top of the one from spack-stack in this case?

Reply:
Give me a few minutes to answer your question please!

Reply:
@climbfuji Thanks for the link; last I had checked spack-stack was still not supported on Derecho so I stuck with the libraries I knew would work. I will see if those work for the SCM build.

Reply:
@climbfuji I wasn't able to get spack-stack to work on Derecho. I receive the following error when attempting to load the spack-stack modules:

```Lmod has detected the following error:  Cannot load module "stack-cray-mpich/8.1.25" without these module(s) loaded:
   libfabric/1.15.2.0

While processing the following module(s):
    Module fullname          Module Filename
    ---------------          ---------------
    stack-cray-mpich/8.1.25  /glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/install/modulefiles/intel/2021.10.0/stack-cray-mpich/8.1.25.lua
    derecho_intel            /glade/derecho/scratch/kavulich/SCM/PR_406/ccpp-scm/scm/etc/modules/derecho_intel.lua
```

Which doesn't make sense to me, because `libfabric/1.15.2.0` is supposed to be loaded *inside* the `stack-cray-mpich/8.1.25` module (and trying to manually load it right before that step doesn't work either). I'll admit I'm a little unclear on the finer details of these modules though, is it possible I'm not loading these in the right order or something?  The modulefile I'm using is below:

```
help([[
This module loads libraries for building the CCPP Single-Column Model on
the CISL machine Derecho (Cray) using Intel-classic-2023.0.0
]])

whatis([===[Loads libraries needed for building the CCPP SCM on Derecho ]===])

load(pathJoin("cmake", os.getenv("cmake_ver") or "3.26.3"))
load(pathJoin("ncarenv", os.getenv("ncarenv_ver") or "23.06"))
load(pathJoin("craype", os.getenv("craype_ver") or "2.7.20"))

prepend_path("MODULEPATH","/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/install/modulefiles/Core")
load("stack-intel/2021.10.0")
load("stack-cray-mpich/8.1.25")
load("stack-python/3.10.8")

load("bacio/2.4.1")
load("sp/2.3.3")
load("w3emc/2.9.2")

setenv("CC","cc")
setenv("FC","ftn")
setenv("CXX","CC")

setenv("CMAKE_C_COMPILER","cc")
setenv("CMAKE_CXX_COMPILER","CC")
setenv("CMAKE_Fortran_COMPILER","ftn")
setenv("CMAKE_Platform","derecho.intel")
```

Reply:
Did you use the "preamble" from spack-stack? In particular, I think you need (see https://spack-stack.readthedocs.io/en/latest/PreConfiguredSites.html#ncar-wyoming-derecho):
```
module purge
# ignore that the sticky module ncarenv/... is not unloaded
export LMOD_TMOD_FIND_FIRST=yes
module load ncarenv/23.09
module use /glade/work/epicufsrt/contrib/spack-stack/derecho/modulefiles
```
and then the module use/load statements for either GNU or Intel.

Reply:
@climbfuji Sorry about that....writing out my last message inspired me to figure out the first issue (you are right I had not done the first step of appending the modulepath for spack stack modules). I am building now and will let you know if these tests succeed.

One thing I noticed is that the `w3emc` libraries are listed without a version in the Derecho spack-stack. This seems odd to me, do you know why this is?

Reply:
> I think that there is still a reliance on some non-standard python packages, like f90nml. Would we still need to create a python environment on top of the one from spack-stack in this case?

@grantfirl It looks like we do need to keep our own python environment, at least for now (the spack-stack environment does not contain f90nml as you expected).

Reply:
It does, `module load py-f90nml`

Reply:
Thanks again @climbfuji. So all the python packages in spack-stack are loaded via modules? I didn't think to look there, it looks like a lot of good packages 👍

Reply:
Yes, all of them are modules and all python packages start with `py-`.

Reply:
You can extend these spack python environments in case packages are missing to load whatever you can as modules from spack-stack, then create a virtual environment (`python3 -m venv venv`) and then install missing packages via pip (`python3 -m pip install NAME`). This way, all the spack-stack Python utilities are used unless there are version conflicts.